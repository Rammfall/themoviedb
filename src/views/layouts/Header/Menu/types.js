import PropTypes from 'prop-types'

export const menuPropTypes = {
  onLogout: PropTypes.func.isRequired
}
