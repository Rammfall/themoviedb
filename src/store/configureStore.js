import { createLogicMiddleware } from 'redux-logic'
import { createStore, applyMiddleware } from 'redux'

import logics from './logics'
import rootReducer from './rootReducer'

const bindMiddleware = (middleware) => {
  if (process.env.NODE_ENV !== 'production') {
    // eslint-disable-next-line
    const { composeWithDevTools } = require('redux-devtools-extension')

    return composeWithDevTools(applyMiddleware(...middleware))
  }

  return applyMiddleware(...middleware)
}

const configureStore = (initialState = {}) => {
  // Add dependencies to pass them to logic functions
  const dependencies = {}

  const logicMiddleware = createLogicMiddleware(logics, dependencies)
  const store = createStore(rootReducer, initialState, bindMiddleware([logicMiddleware]))

  return store
}

export default configureStore
