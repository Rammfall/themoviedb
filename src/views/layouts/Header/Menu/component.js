import React from 'react'
import { Link } from 'react-router-dom'
import { Menu } from 'antd'
import { menuPropTypes } from 'Views/layouts/Header/Menu/types'

const MenuComponent = ({ onLogout }) => (
  <Menu>
    <Menu.Item>
      <Link to='/'>Dashboard</Link>
    </Menu.Item>
    <Menu.Divider />
    <Menu.Item>
      <Link to='/lists'>My Lists</Link>
    </Menu.Item>
    <Menu.Item>
      <Link to='/watchlist'>Watchlist</Link>
    </Menu.Item>
    <Menu.Item>
      <Link to='/favorites'>Favorites</Link>
    </Menu.Item>
    <Menu.Divider />
    <Menu.Item onClick={onLogout}>Logout</Menu.Item>
  </Menu>
)

MenuComponent.propTypes = menuPropTypes

export default MenuComponent
