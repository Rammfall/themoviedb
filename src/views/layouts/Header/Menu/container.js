import React from 'react'
import { connect } from 'react-redux'

import { logoutUser } from 'Store/concepts/session/actions'
import { menuPropTypes } from 'Views/layouts/Header/Menu/types'

import MenuComponent from 'Views/layouts/Header/Menu/component'

const Menu = ({ onLogout }) => (
  <MenuComponent onLogout={onLogout} />
)

Menu.propTypes = menuPropTypes

const MapDispatchToProps = {
  onLogout: logoutUser
}

export { MenuComponent }
export default connect(undefined, MapDispatchToProps)(Menu)
