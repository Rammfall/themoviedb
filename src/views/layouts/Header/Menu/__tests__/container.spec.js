import React from 'react'
import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store'

import diveTo from 'TestUtils/diveToEnzyme'

import Menu, { MenuComponent } from '../container'

describe('Menu', () => {
  const mockStore = configureStore()()

  it('matches snapshot', () => {
    const wrapper = shallow(
      <Menu store={mockStore}
            onLogout={jest.fn}
      />)
    const container = diveTo(wrapper, MenuComponent)

    expect(container).toMatchSnapshot()
  })
})
