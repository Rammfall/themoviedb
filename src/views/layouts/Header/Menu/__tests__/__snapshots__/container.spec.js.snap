// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu matches snapshot 1`] = `
<Menu>
  <MenuItem>
    <Link
      to="/"
    >
      Dashboard
    </Link>
  </MenuItem>
  <Divider
    className=""
    disabled={true}
    style={Object {}}
  />
  <MenuItem>
    <Link
      to="/lists"
    >
      My Lists
    </Link>
  </MenuItem>
  <MenuItem>
    <Link
      to="/watchlist"
    >
      Watchlist
    </Link>
  </MenuItem>
  <MenuItem>
    <Link
      to="/favorites"
    >
      Favorites
    </Link>
  </MenuItem>
  <Divider
    className=""
    disabled={true}
    style={Object {}}
  />
  <MenuItem
    onClick={[Function]}
  >
    Logout
  </MenuItem>
</Menu>
`;
